automation/service/workflow.go

```diff
@@ -582,32 +591,32 @@ func (svc *workflow) Exec(ctx context.Context, workflowID uint64, p types.Workfl
 				}
 			}
 
-			return nil, nil
-		}()
+			if !p.Trace {
+				if t == nil {
+					return nil, WorkflowErrUnknownWorkflowStep()
+				} else if !t.Enabled {
+					return nil, WorkflowErrDisabled()
+				}
+			}
 
-		if err != nil {
-			return
-		}
+			if t != nil {
+				wap.setTrigger(t)
+				p.StepID = t.StepID
+				p.EventType = t.EventType
+				p.ResourceType = t.ResourceType
 
-		if !p.Trace {
-			if t == nil {
-				return WorkflowErrUnknownWorkflowStep()
-			} else if !t.Enabled {
-				return WorkflowErrDisabled()
+				// merge with input from trigger
+				// with trigger input vars are overwritten by input vars
+				p.Input = t.Input.MustMerge(p.Input)
+			} else {
+				p.EventType = "onTrace"
```

Comment:
Can we move to a constant? Magic numbers but as a string :eyes:

Reply:
it would, but this is an old code and not relevant to the PR.